fastplotlib/graphics/selectors/_rectangle_region.py

```diff
@@ -188,91 +89,105 @@ def __init__(
         edge_color: str, array, or tuple
             edge color for the selector, passed to pygfx.Color
 
+        edge_thickness: float, default 8
+            edge thickness
+
+        arrow_keys_modifier: str
+            modifier key that must be pressed to initiate movement using arrow keys, must be one of:
+            "Control", "Shift", "Alt" or ``None``
+
         name: str
             name for this selector graphic
         """
 
+        if not len(selection) == 4 or not len(limits) == 4:
+            raise ValueError()
+
         # lots of very close to zero values etc. so round them
-        bounds = tuple(map(round, bounds))
+        selection = tuple(map(round, selection))
         limits = tuple(map(round, limits))
-        origin = tuple(map(round, origin))
 
-        Graphic.__init__(self, name=name)
+        self._parent = parent
+        self._limits = np.asarray(limits)
+        self._resizable = resizable
 
-        self.parent = parent
+        selection = np.asarray(selection)
 
         # world object for this will be a group
         # basic mesh for the fill area of the selector
         # line for each edge of the selector
         group = pygfx.Group()
-        self._set_world_object(group)
 
-        xmin, xmax, ymin, ymax = bounds
+        xmin, xmax, ymin, ymax = selection
 
         width = xmax - xmin
         height = ymax - ymin
 
+        if width < 0 or height < 0:
+            raise ValueError()
+
         self.fill = pygfx.Mesh(
             pygfx.box_geometry(width, height, 1),
             pygfx.MeshBasicMaterial(color=pygfx.Color(fill_color), pick_write=True),
         )
 
-        self.fill.position.set(*origin, -2)
-        self.world_object.add(self.fill)
+        self.fill.world.position = (0, 0, -2)
 
-        # position data for the left edge line
+        group.add(self.fill)
+
+        #position data for the left edge line
         left_line_data = np.array(
             [
-                [origin[0], (-height / 2) + origin[1], 0.5],
-                [origin[0], (height / 2) + origin[1], 0.5],
+                [0, -width / 2, 0],
+                [0, width / 2, 0],
```

Comment:
can this now just be `[xmin, ymin, 0], [xmin, ymax, 0]` ? And similarly for the other lines